x/oracle/keeper/genesis.go

```diff
@@ -21,15 +21,26 @@ func (k Keeper) IterateAllHistoricPrices(
 		k.cdc.MustUnmarshal(iter.Value(), &decProto)
 		denom, blockNum := types.ParseDenomAndBlockFromKey(iter.Key(), types.KeyPrefixHistoricPrice)
 		historicPrice := types.Price{
-			ExchangeRateTuple: types.ExchangeRateTuple{ExchangeRate: decProto.Dec, Denom: denom},
-			BlockNum:          blockNum,
+			ExchangeRate: &sdk.DecCoin{Denom: denom, Amount: decProto.Dec},
+			BlockNum:     blockNum,
 		}
 		if handler(historicPrice) {
 			break
 		}
 	}
 }
 
+// AllHistoricPrices is a helper function that collects and returns all
```

Comment:
None of these functions are exclusive to genesis anymore but I don't see a good reason to move them anywhere.